main.cpp

```diff
@@ -363,17 +197,20 @@ int main(int argc, char ** argv)
 			}
 		}
 
+		if (status.is_finished) goto done;
```

Comment:
If it's simple return or flow controlling condition, it's ok.
This style also often appeared in libtorrent like [here](https://github.com/arvidn/libtorrent/blob/master/src/allocator.cpp#L80).